adapters/pubmatic/pubmatic.go

```diff
@@ -331,6 +329,11 @@ func (a *PubmaticAdapter) MakeRequests(request *openrtb2.BidRequest, reqInfo *ad
 		}
 	}
 
+	// If all the requests are invalid, Call to adaptor is skipped
+	if len(request.Imp) == 0 {
```

Comment:
Moved this check up in code to avoid other work which would otherwise be wasted.